### PR TITLE
Add blog post about Rails upgrade

### PR DESCRIPTION
**Pull request by TatianaPan**
With this PR I would like to share my experience in doing Rails upgrades. Learning described approach helped me a lot - I thought it might be useful for other people as well.

### CODE REVIEW

**Review by crackofdusk — COMMENTED** [review]
Nice! I like that you give concrete examples.

What do you think about recapping your steps in a list at the end of the post?

File: content/posts/rails-upgrade-strategy.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+One more example. When I once tried to do a Rails upgrade without updating dependencies in advance, all tests were broken. The error message was very weird and not helpful at all. Google and Stackoverflow also did not know about such a problem. I changed my approach and started to update gems one by one. It turned out that the `database-cleaner` gem, that our application uses, introduced some breaking changes since our last update (it actually became a set of gems depending on the used database adapters). In our case I had to install a new gem `database-cleaner_mongoid` and do some configuration changes. The problem was solved!
+
+Another option to perform Rails upgrades smoothly is to keep your dependencies up to date. You can do it easily by installing [dependabot](https://docs.github.com/en/code-security/dependabot). It will run your pipeline against any new gem updates and you will be able to catch breaking changes early.
+
+Often before the Rails upgrade you would also like to upgrade to a newer Ruby version. Here the same principle is applied: if the upgrade does not work out of the box and you get errors, try to update first those dependencies that block the Ruby upgrade.
```
[the remark this answers is not in the document]
Comment (TatianaPan) [reply content/posts/rails-upgrade-strategy.md:25]: Thanks @pinefoambath ! It's a very good point why to do an upgrade at all, I added this paragraph to my text. But I am not sure about proposing to perform Rails upgrades as one of our services. I thought the main target group of this post would be developers. But maybe it still makes sense to advertise such a service? What do you think @crackofdusk ?

**Comment by TatianaPan** [comment]
@crackofdusk @pinefoambath thank you very much for your review! I adjusted text according to your comments. Could you please have a look again? Feel free to correct my English, add missing articles etc.. ;)

File: content/posts/rails-upgrade-strategy.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+Here once again all steps recapped:
+
+1. Upgrade Ruby version to the latest compatible with the desired Rails version
+2. If this does not work, individually upgrade gems blocking Ruby update, and try again
+3. Update Rails to the next patch version
```
Comment (crackofdusk) [inline content/posts/rails-upgrade-strategy.md:31]:
```suggestion
3. Update Rails to the most recent patch version of the currently used release series
```

File: content/posts/rails-upgrade-strategy.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+4. Run `bundle outdated` and update one by one all outdated gems
```
Comment (crackofdusk) [inline content/posts/rails-upgrade-strategy.md:32]:
```suggestion
4. Run `bundle outdated` and update all outdated gems one by one
```

File: content/posts/rails-upgrade-strategy.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+5. Perform Rails upgrade to a minor or a major version
```
Comment (crackofdusk) [inline content/posts/rails-upgrade-strategy.md:33]:
```suggestion
5. Perform the Rails upgrade to a minor or a major version following the [Rails upgrade guide](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html)
```

File: content/posts/rails-upgrade-strategy.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+---
+title: "How-To: Rails upgrade strategy"
+date: 2022-05-25
```
Comment (crackofdusk) [inline content/posts/rails-upgrade-strategy.md:3]: Don't forget to update the publication date before merging.

**Review by crackofdusk — APPROVED** [review]

**Pull request merged**